src/restconfig/src/main/java/org/geoserver/rest/catalog/CoverageStoreFileController.java

```diff
@@ -453,4 +459,27 @@ protected Resource findPrimaryFile(Resource directory, String format) {
         return handleFileUpload(
                 storeName, workspaceName, filename, method, format, directory, request);
     }
+
+    /** Return the remote URL provided in the request. */
+    protected URL handleRemoteUrl(HttpServletRequest request) {
+
+        try {
+            // get the URL to be harvested
+            final String stringURL;
+            try (InputStream inStream = request.getInputStream()) {
+                stringURL = IOUtils.getStringFromStream(inStream);
```

Comment:
A bit dangerous, there is no indication of the charset that might have been used.
Better to get the reader from the request instead of the stream:

```
   test = IOUtils.toString(request.getReader()); 
```

Using commons-io IOUtils.

Reply:
Done